.github/workflows/static-analyzers.yml

```diff
@@ -13,3 +13,39 @@ jobs:
     - uses: actions/checkout@v2
     - name: Run cpplint
       run: cpplint --counting=detailed --recursive examples benchmarks src tests
+
+  clang-tidy:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    container: helics/buildenv:clang-extra-tools
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run clang-tidy on changed files
+      shell: bash
+      run: |
+        FILES_URL="$(jq -r '.pull_request._links.self.href' "$GITHUB_EVENT_PATH")/files"
```

Comment:
would it be appropriate to put this in a separate script?  looks like it might be getting a bit long for having in a yml file or are there aspects of it that require being in the yml file